modules/caddyhttp/caddyauth/caddyauth.go

```diff
@@ -78,6 +78,9 @@ func (a Authentication) ServeHTTP(w http.ResponseWriter, r *http.Request, next c
 
 	repl := r.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
 	repl.Set("http.authentication.user.id", user.ID)
+	for k, v := range user.Metadata {
+		repl.Set("http.authentication.user.metadata."+k, v)
```

Comment:
```suggestion
		repl.Set("http.authentication.user."+k, v)
```

@roblabla Would this be OK with you? I can't think of a reason we need to separate out the metadata fields from anything else really.

Reply:
Part of me wonders if we even need the `.authentication` namespace at all, wouldn't it be obvious that a user is from some kind of authentication? i.e. we could do just `http.user.*` - what do you think?